### ISSUE TITLE
Extensions to the carbon for products wizard

### ISSUE THREAD

**Issue by MayaEapen** [carbon-design-system/ibm-products#2930]
# Extensions to the carbon for products wizard

## Brief description
The HPM design team is proposing to add the option to add error icons to the progress indicator of the Carbon for IBM Products wizard when the step contains an error. 

These are functionality enhancements.

## Details

**Existing implementation** 
- https://pages.github.ibm.com/cdai-design/pal/patterns/creation-flows/usage
- https://carbon-for-ibm-products.netlify.app/?path=/story/ibm-products-patterns-create-flows-createfullpage--create-full-page

**Proposed implementation** 
![Screenshot 2023-04-26 at 11 58 00](https://user-images.githubusercontent.com/131867065/234541338-d49e13d5-21cf-4f42-b2e8-86ad642934bc.png)

### Links to other materials

e.g. sketch files, icons, images, prototypes

## Design Maintainer(s)

- Owner/main maintainer(s): Maya Eapen, Marie Made, Antoni
- Second/backup maintainer(s)

## Tasks

Before starting work on this epic, please review and complete the following.

- [ ] Initial review of design/existing code

### Working in Carbon for IBM Products package

- [ ] Have you reviewed our
      [contribution guidelines](https://github.com/carbon-design-system/ibm-cloud-cognitive/blob/master/.github/CONTRIBUTING.md)?
- [ ] Have you noted our
      [code guidelines](https://github.com/carbon-design-system/ibm-cloud-cognitive/blob/master/docs/CODE_GUIDELINES.md)?
- [ ] Did you use the CLI to scaffold your new component?

**Comment by baraaabdelsalam** [comment]
FYI: This work has been picked up and is ongoing, but there is no pull request yet.

**Comment by baraaabdelsalam** [comment]
@elycheea
I used this issue as a reference in my pull-request... is that sufficient enough? or should I create a dev-issue instead? Also are there any changes required from the design team or is it sufficient enough that its all showing up in the storybook examples that I created? Note that this is already a feature that is designed in Carbon and supported by the ProgressIndicator but wasn't used by the C4P Create flow patterns.

**Comment by baraaabdelsalam** [comment]
@elycheea @lee-chase 

I used this issue as a reference in my pull-request... should I create a dev-issue and reference it in that Pull-request instead? Also are there any changes or documentation updates required from the design team? its all showing up in the storybook examples that I created and this is already a feature that is designed in Carbon and supported by the ProgressIndicator but wasn't used by the C4P Create flow patterns.
